From 9509e811459ded31e3091af8d50ed81f4a15ba4c Mon Sep 17 00:00:00 2001
From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Tue, 6 Jul 2021 18:31:54 +0530
Subject: [PATCH 3/4] pm: microg-spoof: Add null check

* Usually happens in work profile without microG

PackageManagerService.MICROG_SPOOF_SIGNATURE: java.lang.NullPointerException: Attempt to write to field 'android.content.pm.Signature[] android.content.pm.PackageInfo.signatures' on a null object reference
PackageManagerService.MICROG_SPOOF_SIGNATURE: 	at com.android.server.pm.PackageManagerService.mayFakeSignature(PackageManagerService.java:4460)

Change-Id: I8e582e18a02ce7a3822eb468cd4a7d576d882f3b
---
 services/core/java/com/android/server/pm/ComputerEngine.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/pm/ComputerEngine.java b/services/core/java/com/android/server/pm/ComputerEngine.java
index 44c8cb4dd742..4883a1c5341d 100644
--- a/services/core/java/com/android/server/pm/ComputerEngine.java
+++ b/services/core/java/com/android/server/pm/ComputerEngine.java
@@ -1617,7 +1617,9 @@ public class ComputerEngine implements Computer {
                 if (DEBUG_PACKAGE_INFO) {
                     Log.v(TAG, "Spoofing signature for microG");
                 }
-                pi.signatures = new Signature[] {new Signature(MICROG_FAKE_SIGNATURE)};
+                if (pi != null) {
+                    pi.signatures = new Signature[] {new Signature(MICROG_FAKE_SIGNATURE)};
+                }
             }
         } catch (Throwable t) {
             // We should never die because of any failures, this is system code!
-- 
2.34.1

